content/docs/guides/continuous-delivery/pulumi-kubernetes-operator.md

```diff
@@ -0,0 +1,207 @@
+access_token = core.v1.Secret("accesstoken", string_data={ "access_token": pulumi_access_token })
+
+# Create an NGINX deployment in-cluster.
+my_stack = apiextensions.CustomResource("my-stack",
+    api_version="pulumi.com/v1alpha1",
+    kind="Stack",
+    spec={
+        "access_token_secret": access_token.metadata["name"],
+        "stack": "<YOUR_ORG>/nginx/dev",
+        "init_on_create": True,
+        "project_repo": "https://github.com/metral/pulumi-nginx",
+        "commit": "2b0889718d3e63feeb6079ccd5e4488d8601e353",
+        "destroy_on_finalize": True,
+    }
+)
+```
+
+{{% /choosable %}}
+{{% choosable language csharp %}}
+
+```csharp
+Coming Soon
+```
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+```go
+Coming Soon
+```
+
+{{% /choosable %}}
+{{% /chooser %}}
+
+[stacks-use-kubectl]: https://github.com/pulumi/pulumi-kubernetes-operator/blob/master/docs/create-stacks-using-kubectl.md
+
+### Stack Settings
+
+Stack CustomResources can have the following properties configured that
+influence the Stack update run:
+
+- The first is the access token secret, better known as `PULUMI_ACCESS_TOKEN`, which is required to authenticate with pulumi.com in order to
+  perform the update. You can create a new Pulumi access token specifically for your
+  CI/CD job on your [Pulumi Account page](https://app.pulumi.com/account/tokens).
+- Environment variables to make available to the Stack that are sourced from Kubernetes ConfigMaps and/or Secrets. Examples include
+  cloud provider credentials, and other application settings.
+- Pulumi Stack configs and secrets that can complement or override settings
+  in the repo for use within the Stack.
+- Project repo settings like the repo URL, the commit to deploy, and a repo
+  access token for private repos or rate-limiting.
+- And, lifecyle control such as creating the stack if it does not exist,
+  issuing a refresh before the update, and destroying the Stack's resources
+  and stack itself upon deletion of the CR.
+
+### Extended Examples
+
+Check out how to [manage a Kubernetes Blue/Green Deployment][blue-green-demo]
+using the Operator and a Stack CR.
+
+[blue-green-demo]: https://github.com/pulumi/pulumi-kubernetes-operator/tree/master/examples/blue-green
+
+## Concurrency
+
+When using the operator to continuously deploy your Pulumi stacks, you may run into a problem. What
+happens if multiple reconcilation loops are run for the same commit in succession?
+
+Operators by definition will invoke a reconcilation loop for the creation, update, or deletion of a Stack CR.
+
+Pulumi blocks any stack updates while one is already in progress. (To avoid conflicting resource
+updates or corrupting resource state.) So the stack and its resources won't be harmed by the
+concurrent update. By default, the operator is informed if a conflicting
+update is already in progress and will not spawn another reconcilation loop to
+give the current update in progress sole control of the update run.
```

Comment:
```suggestion
To avoid conflicting resource updates or corrupting resource state, Pulumi only
runs one update at a time per stack. By default, the operator checks for updates
already in progress, and will not spawn another reconciliation loop if one is already
running.
```